File: Desktop/View/Table/Header/SortableHeader.cs
using System.Collections.Generic;
using System.Linq;

namespace Desktop.View.Table.Header
{
    public class SortableHeader
    {
        private List<SortableButton> Buttons { get; } = new List<SortableButton>();

        // Add Button To Button List
        public void Add(SortableButton button) => Buttons.Add(button);

        public void Update(SortableButton button)
        {
            // Reset States Of Buttons != To pressed button
            foreach (var b in Buttons.Where(b => !Equals(button, b)))
            {
                b.State = SortableButton.ButtonState.None;
            }
        }
    }
}

File: Mobile/app/src/main/java/com/mauzerov/mobile/views/SortableHeader.kt
package com.mauzerov.mobile.views

class SortableHeader {
    private var buttons = mutableListOf<SortableButton>()

    operator fun plusAssign(button : SortableButton) {
        buttons.add(button);
    }

    fun update(main : SortableButton) {
        for (button in buttons) {
            if (main === button)
               continue

            button.state = SortableButton.ButtonState.None;
        }
    }
}